arcade/arcade/core/errors.py

```diff
@@ -38,15 +40,26 @@ class ToolDefinitionError(ToolError):
 
 
 class ToolRuntimeError(RuntimeError):
+    _can_retry: bool = False
```

Comment:
I think `can_retry` should be a property of the error, so we can use it in different types of errors when needed.